department-of-veterans-affairs/vets-api
# Retryable/Idempotent SubmitDependentPdfJob

## Summary
- Utilized Sidekiq retry capabilities to auto retry job
- Ensure files are only uploaded to VBMS once utilizing completed_at timestamp on PersistentAttachments and a basic array logging form names for the base forms.
- Completely rewrite `submit_dependents_pdf_job_spec.rb` and `dependency_claim_spec.rb`

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/65144
- https://github.com/department-of-veterans-affairs/vets-api/pull/13907


## Testing done
- Create a claim via `/view-change-dependents/add-remove-form-21-686c/`
- Check in `/sidekiq` admin console that claim will retry if there is an error (locally it will fail always)

## Screenshots
![Screenshot 2023-09-19 at 17 02 31](https://github.com/department-of-veterans-affairs/vets-api/assets/583973/83ac0541-0c49-49ce-941e-3865e81cdaad)



## What areas of the site does it impact?
None are functionally impacted, this should only assist in background jobs being handled better after errors

## Acceptance criteria


## Next Steps
Merge https://github.com/department-of-veterans-affairs/vets-api/pull/13907
